var/spack/repos/builtin/packages/icedtea/package.py

```diff
@@ -139,6 +141,19 @@ class Icedtea(AutotoolsPackage):
              url='http://icedtea.wildebeest.org/download/drops/icedtea8/3.4.0/shenandoah.tar.xz',
              when='@3.4.0')
 
+    # FIXME:
```

Comment:
(1) and (2) have nothing in common. (1) is "you can't extend virtual packages" and (2) is "you can't extend multiple packages". Although, maybe (1) is another symptom of (2)? Anyway, (1) will be really important if we ever make `python` a virtual package.

(3) is not strictly required. You're right, Java developers rarely suggest installing `.jar` files directly to `lib/ext`. But it does work, so if it isn't too much work, I might as well support it. I'm sure there's an easy way to do this, I just don't know how.

Reply:
> (1) and (2) have nothing in common. (1) is "you can't extend virtual packages" and (2) is "you can't extend multiple packages"

What I mean is: if you could extend multiple packages, my understanding is you would say 

```
extends('jdk')
extends('icedtea')
```

And you wouldn't need to add `extends('java')`.

That being said, the primary benefits of marking a package as an extension of some other package is:

* The dependent can be activated in the prefix of the extendee
* You can manage centralized state (e.g. a merged `.pth` file for Python)

Since Java doesn't need to manage centralized state for dependents, only the first benefit would apply. As mentioned in https://github.com/spack/spack/pull/8613#issuecomment-412242516, you can implement whatever customized merging logic you want without waiting for the extension issues to be resolved.

Reply:
I'm doing a bad job of inferring why you want this so I'll be direct: why is the ability to extend multiple packages important here? Could you give me an example of what you would write if this were possible?

Reply:
See #987. There are already multiple packages in Spack that extend multiple languages. For example, some packages have both Python and Perl libraries.